internal/server/singleprocess/poll_application.go

```diff
@@ -116,15 +116,15 @@ func (a *applicationPoll) buildPollJobs(
 	if err != nil {
 		// If the deployment isn't found, it's ok
 		s, ok := status.FromError(err)
-		if !ok || s.Code() != codes.NotFound {
+		if !ok && s.Code() != codes.NotFound {
```

Comment:
Hm, this doesn't seem right to me. I pulled up the `FromError` impl to try to figure out what we're trying to do here: https://github.com/grpc/grpc-go/blob/41e044e1c82fcf6a5801d6cbd7ecf952505eecb1/status/status.go#L81

I think we can just change this to `status.Code`? (https://pkg.go.dev/google.golang.org/grpc/status#Code)

On L116:

```
if err != nil && status.Code(err) != codes.NotFound {
```

If there was an error and it wasn't a not found error, then we have to return an error.

Is that correct?

Reply:
Q1: I believe there are tests under [poll_tests.go](https://github.com/hashicorp/waypoint/blob/main/internal/server/singleprocess/poll_test.go#L462) for this?

Q2: Good call on the error check updates! Change pushed.

Q3: We poll status per project, so if we exit here on an error because one app has no deployments or releases to poll, then returning an error would cause us to exit with error out of the entire polling loop. We do check on line 126 for nil deployments and releases and return `nil, nil` to not queue a job, but to still continue polling any additional apps in the project.

Reply:
> Q1: I believe there are tests under poll_tests.go for this?

Ah, so there are... we should probably move those to `poll_application_test.go`. And I would expect a fix like this to have a regression test associated with it. We can add those later though in the interest of just getting it working.